client/homebrew/pages/sharePage/sharePage.jsx

```diff
@@ -79,6 +79,21 @@ const SharePage = createClass({
 		</div>;
 	},
 
+	renderPrint : function(){
+		if(!this.props.brew.shareId) return;
+		return <PrintLink shareId={this.processShareId()} />;
+	},
+
+	renderSourceDropdown : function(){
+		if(!this.props.brew.shareId) return;
+		return <Nav.item icon='fas fa-code' color='red' className='source'
+						onMouseEnter={()=>this.handleDropdown(true)}
+						onMouseLeave={()=>this.handleDropdown(false)}>
+						source
+						{this.renderDropdown()}
+					</Nav.item>;
+	},
```

Comment:
Probably doesn't need to even be in separate functions up here anymore. Since they both use the same check, just put the check directly in the render function.

```
      {this.props.brew.shareId  &&
        <PrintLink shareId={this.processShareId()} />
	<Nav.item icon='fas fa-code' color='red' className='source'
		onMouseEnter={()=>this.handleDropdown(true)}
		onMouseLeave={()=>this.handleDropdown(false)}>
		source
		{this.renderDropdown()}
	</Nav.item>
      }
```